pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -122,6 +122,7 @@ type ApplicationBackupController struct {
 	recorder             record.EventRecorder
 	resourceCollector    resourcecollector.ResourceCollector
 	backupAdminNamespace string
+	terminationChannels  []chan bool
```

Comment:
The controller handles different ApplicationBackup CRs at the same time. You need to maintain a separate terminationChannels slice for each applicationBackup CR. 
```
terminationChannelsMap map[string][]chan bool // key is the ApplicationBackup.UID
```

Reply:
Good catch @adityadani , will chnage it to a map with value as this slice..

Reply:
@adityadani  I have addressed this comment in the latest PR that is raised against master.